wiring/inc/spark_wiring_ble.h

```diff
@@ -0,0 +1,547 @@
+public:
+    BleAddress& operator=(hal_ble_addr_t addr) {
+        this->addr_type = addr.addr_type;
+        memcpy(this->addr, addr.addr, BLE_SIG_ADDR_LEN);
+        return *this;
+    }
+
+    uint8_t operator[](uint8_t i) const {
+        if (i >= BLE_SIG_ADDR_LEN) {
+            return 0;
+        }
+        return addr[i];
+    }
+
+    bool operator==(const BleAddress& addr) const {
+        if (this->addr_type == addr.addr_type && !memcmp(this->addr, addr.addr, BLE_SIG_ADDR_LEN)) {
+            return true;
+        }
+        return false;
+    }
+};
+static_assert(std::is_pod<BleAddress>::value, "BleAddress is not a POD struct");
+
+class BleAdvertisingParams : public hal_ble_adv_params_t {
+};
+static_assert(std::is_pod<BleAdvertisingParams>::value, "BleAdvertisingParams is not a POD struct");
+
+class BleConnectionParams : public hal_ble_conn_params_t {
+};
+static_assert(std::is_pod<BleConnectionParams>::value, "BleConnectionParams is not a POD struct");
+
+class BleScanParams : public hal_ble_scan_params_t {
+};
+static_assert(std::is_pod<BleScanParams>::value, "BleScanParams is not a POD struct");
+
+class BleCharacteristicHandles : public hal_ble_char_handles_t {
+public:
+    BleCharacteristicHandles& operator=(const hal_ble_char_handles_t& halHandles) {
+        this->size = halHandles.size;
+        this->decl_handle = halHandles.decl_handle;
+        this->value_handle = halHandles.value_handle;
+        this->user_desc_handle = halHandles.user_desc_handle;
+        this->cccd_handle = halHandles.cccd_handle;
+        this->sccd_handle = halHandles.sccd_handle;
+        return *this;
+    }
+};
+static_assert(std::is_pod<BleCharacteristicHandles>::value, "BleCharacteristicHandles is not a POD struct");
+
+
+class BleUuid {
+public:
+    BleUuid()
+            : uuid_() {
+    }
+    BleUuid(const BleUuid& uuid);
+    BleUuid(const uint8_t* uuid128, BleUuidOrder order = BleUuidOrder::LSB);
+    BleUuid(uint16_t uuid16);
+    BleUuid(const uint8_t* uuid128, uint16_t uuid16, BleUuidOrder order = BleUuidOrder::LSB);
+    BleUuid(const String& uuid);
+    BleUuid(const char* uuid);
+    ~BleUuid() = default;
+
+    bool isValid() const;
+
+    BleUuidType type() const {
+        if (uuid_.type == BLE_UUID_TYPE_16BIT || uuid_.type == BLE_UUID_TYPE_128BIT_SHORTED) {
+            return BleUuidType::SHORT;
+        } else {
+            return BleUuidType::LONG;
+        }
+    }
+
+    hal_ble_uuid_t UUID() const {
+        return uuid_;
+    }
+
+    hal_ble_uuid_t& UUID() {
+        return uuid_;
+    }
+
+    uint16_t shorted() const {
+        return uuid_.uuid16;
+    }
+
+    void full(uint8_t uuid128[BLE_SIG_UUID_128BIT_LEN]) const {
+        memcpy(uuid128, uuid_.uuid128, BLE_SIG_UUID_128BIT_LEN);
+    }
+
+    const uint8_t* full() const {
+        return uuid_.uuid128;
+    }
+
+    bool operator==(const BleUuid& uuid) const;
+
+private:
+    hal_ble_uuid_t uuid_;
+};
+
+
+class iBeacon {
```

Comment:
Let's move this into a separate header, this is not a part of the core functionality.